cocos2d/core/CCActionManager.js

```diff
@@ -339,7 +339,8 @@ cc.ActionManager = cc.Class.extend(/** @lends cc.ActionManager# */{
             //this._currentTargetSalvaged = false;
             if (!locCurrTarget.paused) {
                 // The 'actions' CCMutableArray may change while inside this loop.
-                for (locCurrTarget.actionIndex = 0; locCurrTarget.actionIndex < locCurrTarget.actions.length;
+                for (locCurrTarget.actionIndex = 0;
+                     locCurrTarget.actionIndex < (locCurrTarget.actions ? locCurrTarget.actions.length : 0);
```

Comment:
@VisualSJ 
@pandamicro

three questions for the loop:
1,as said in the comments: ""The 'actions' CCMutableArray may change while inside this loop"". so is it safe to make loop for a changing Array for getting the right value from the Array?
a test case that will print the result maybe you want to get:
```
var ii = [1,2,3,4,5,6,7,8,9,0];
for (var i = 0; i < ii.length; i++) {console.log(ii[i]); if (ii[i]==5) ii.splice(1, 1);}
```
2,kindly consider that if it is in the inner loop then actions changed, is the logic right or needed or will lead to collapse？
for example, element.actions[3] = null;

3, what thing will lead to the change for actions? describing it clearly is good for reader. and is it reasonable？